### PR TITLE
[PendingArmReview] Fixed Settings API

### PR DESCRIPTION
**Pull request by taklei**
Fixing two bugs - 
1. The setting kind expected in the controller is DataExportSetting**s** and not DataExportSetting
2. The auto generated code doesn't work properly because of a polymorphic issue - moved the discriminator part to the class that actually has the property kind 

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by anthony-c-martin** [comment]
I've signed off for ARM - @taklei has confirmed that this is a correction, and that the existing SDK client does not work against their service. So I do not consider this a breaking change.

**Review by nitsi — APPROVED** [review]

**Review by ChenTanyi — APPROVED** [review]

File: specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "setting": {
       "id": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/providers/Microsoft.Security/settings/MCAS",
       "name": "MCAS",
-      "kind": "DataExportSetting",
+      "kind": "DataExportSettings",
       "type": "Microsoft.Security/settings",
       "properties": {
```
Comment (raych1) [inline specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json:11]: `properties` doesn't exist in the definition and needs to be removed.

File: specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "setting": {
       "id": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/providers/Microsoft.Security/settings/MCAS",
       "name": "MCAS",
-      "kind": "DataExportSetting",
+      "kind": "DataExportSettings",
       "type": "Microsoft.Security/settings",
```
Comment (raych1) [inline specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json:10]: `type` is `readonly` property and cannot be sent in the request. pls remove it.

File: specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json (hunk 1 above)
Comment (taklei) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json:11]: "properties" exists in the DataExportSettings definition which inherits from Setting.
Technically: we want to pass the derived class (DataExportSettings ) and Setting is our base class. 
Is there a way to overcome this error? Maybe adding it in the definition this way:

    "Setting": {
      "type": "object",
      "description": "Represents a security setting in Azure Security Center.",
      "properties": {},
      "allOf": [
        {
          "$ref": "#/definitions/SettingResource"
        }
      ]
    }

File: specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json (hunk 2 above)
Comment (taklei) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json:10]: Okay, thanks

File: specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json (hunk 1 above)
Comment (raych1) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json:11]: yes, leave it in base class.

Comment (taklei) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json:10]: Added the   "properties": {}, to the Setting definition and "additionalProperties": true to the SettingResource
Didn't solve the issue :(

Comment (taklei) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Settings/UpdateSetting_example.json:8]: Yay fixed it!

**Pull request merged**